indexer/indexer_tests/trie_test.cpp

```diff
@@ -83,74 +84,82 @@ string DebugPrint(KeyValuePair const & p)
 
 struct KeyValuePairBackInserter
 {
-  vector<KeyValuePair> m_v;
   template <class TString>
-  void operator()(TString const & s, trie::FixedSizeValueReader<4>::ValueType const & rawValue)
+  void operator()(TString const & s, uint32_t const & value)
   {
-    uint32_t value;
-    memcpy(&value, &rawValue, 4);
     m_v.push_back(KeyValuePair(s, value));
   }
+
+  vector<KeyValuePair> m_v;
 };
 
-struct MaxValueCalc
+// The SingleValueSerializer and ValueList classes are similar to
+// those in indexer/string_file_values.hpp but that file
+// is not included to avoid coding_tests's dependency from indexer.
```

Comment:
Coding tests should NOT depend on indexer. It can be easier to move some class from indexer to coding than introduce this dependency.

Coding library is everything about ‘generic’ data encoding/decoding/converting etc. plus we store our file access code there too.

Indexer library is everything about our features and geo related algorithms to access these features.